File: solution/c#/Day13/Day13/Blog.cs
namespace Day13
{
    public class Article
    {
        private readonly string _name;
        private readonly string _content;
        public List<Comment> Comments { get; }

        public Article(string name, string content)
        {
            _name = name;
            _content = content;
            Comments = new List<Comment>();
        }

        private void AddComment(
            string text,
            string author,
            DateOnly creationDate)
        {
            var comment = new Comment(text, author, creationDate);
            if (Comments.Contains(comment))
            {
                throw new CommentAlreadyExistException();
            }
            else Comments.Add(comment);
        }

        public void AddComment(string text, string author)
            => AddComment(text, author, DateOnly.FromDateTime(DateTime.Now));
    }

    public record Comment(string Text, string Author, DateOnly CreationDate);

    public class CommentAlreadyExistException : ArgumentException;
}

File: exercise/kotlin/day04/src/main/kotlin/blog/Blog.kt
package blog

import java.time.LocalDate

class Article(private val name: String, private val content: String) {
    private val comments: MutableList<Comment> = mutableListOf()

    private fun addComment(
        text: String,
        author: String,
        creationDate: LocalDate
    ) {
        val comment = Comment(text, author, creationDate)
        if (comments.contains(comment)) {
            throw CommentAlreadyExistException()
        } else comments.add(comment)
    }

    fun addComment(text: String, author: String) = addComment(text, author, LocalDate.now())
    fun getComments(): List<Comment> = comments
}


data class Comment(val text: String, val author: String, val creationDate: LocalDate)
class CommentAlreadyExistException : Exception()